apps/distgo/cmd/dist/scheduler.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2016 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dist
+
+import (
+	"sort"
+
+	"github.com/pkg/errors"
+
+	"github.com/palantir/godel/apps/distgo/params"
+)
+
+func Schedule(specsWithDeps []params.ProductBuildSpecWithDeps) ([]params.ProductBuildSpecWithDeps, error) {
```

Comment:
`Schedule` isn't the most descriptive name here -- when I see `dist.Schedule`, I would expect such a function to actually schedule actions rather than just ordering input.

Maybe something like `OrderSpecs` or `OrderBuildSpecs` or `SortBuildSpecsTopologically`? Comment describing what the function does would also be helpful.